cdap-app-templates/cdap-etl/cdap-data-pipeline/src/test/java/co/cask/cdap/datapipeline/DataPipelineTest.java

```diff
@@ -1304,6 +1308,222 @@ private void testSecureStorePipeline(Engine engine, String prefix) throws Except
     Assert.assertNotNull(getDataset(prefix + "MockSecureSinkDataset").get());
   }
 
+  @Test
+  public void testExternalDatasetTrackingMR() throws Exception {
```

Comment:
That test is not a good test. The code path is covered by the tests for the config class itself and should not need a very expensive test that builds jars and tries to deploy an app.